src/app/core/entity/schema/entity-schema-field.ts

```diff
@@ -26,6 +26,13 @@ import { EntityReferenceRole } from "../../basic-datatypes/entity/entity-referen
  * `@DatabaseField(fieldConfig: EntitySchemaField)`
  */
 export interface EntitySchemaField {
+  /**
+   * The key (property name) on the entity object for this field.
+   *
+   * Do not change after initial field creation!
+   */
+  readonly id: string;
```

Comment:
"name" to me sounds like something that might be a "human-readable" name as well instead of a technical, internal name. "key" or "id" (which we also use for filters, for example) makes that more intuitively clear, I think.